File: VisualizerJava/app/src/main/java/com/example/visualizerjava/MainActivity.java
package com.example.visualizerjava;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import android.media.AudioRecord;
import android.os.AsyncTask;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

import java.util.Objects;

public class MainActivity extends AppCompatActivity {

    private MainActivity.Record _record;
    boolean isRecording = false;
    Button button;
    TextView textView0;
    TextView textView1;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        textView0 = findViewById(R.id.t0);
        textView1 = findViewById(R.id.t1);

        button = findViewById(R.id.buttonStart);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if(isRecording){
                    stopRecord();
                }else {
                    startRecord();
                }

            }
        });
    }

    private void stopRecord(){
        isRecording =false;
        button.setText("start");

        if(_record != null) {
            _record.cancel(true);
        }
    }
    private  void startRecord(){
        isRecording =true;
        button.setText("stop");

        this._record = new MainActivity.Record();
        _record.execute();
    }

    public final class Record extends AsyncTask {
        @Nullable
        protected Void doInBackground() {
            int sampleRate = 44100;
            int minBufferSize = AudioRecord.getMinBufferSize(sampleRate, 16, 2) * 2;
            if (minBufferSize < 0) {
                return null;
            } else {
                AudioRecord audioRecord = new AudioRecord(1, sampleRate, 12, 2, minBufferSize);
                int sec = 1;
                short[] buffer = new short[sampleRate * 2 * 1 * sec];
                audioRecord.startRecording();

                try {
                    while (isRecording) {
                        int readSize = audioRecord.read(buffer, 0, minBufferSize);
                        if (readSize < 0) {
                            break;
                        }

                        if (readSize != 0) {
                            boolean var9 = false;
                            if (buffer.length == 0) {
                                break;
                            }
                            if (buffer[0] < 1000 && buffer[0] > -1000) {
                                textView0.setText("ok");
                            } else {
                                textView0.setText("no");
                            }
                            textView1.setText(String.valueOf(buffer[0]));
                        }
                    }
                } finally {
                    audioRecord.stop();
                    audioRecord.release();
                }

                return null;
            }
        }

        // $FF: synthetic method
        // $FF: bridge method
        public Object doInBackground(Object[] var1) {
            return this.doInBackground();
        }
    }
}

File: Visualizer/app/src/main/java/com/example/visualizer/MainActivity.kt
package com.example.visualizer

import android.graphics.Color
import android.media.AudioFormat
import android.media.AudioRecord
import android.media.MediaRecorder
import android.os.AsyncTask
import android.os.Bundle
import android.view.SurfaceView
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    var _record: Record? = null
    var _isRecording = false
    var _button: Button? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        _button = findViewById(R.id.buttonStart) as Button
        _button?.setOnClickListener {
            if(_isRecording)
                stopRecord()
            else
                doRecord()
        }
    }

    override fun onBackPressed() {
        super.onBackPressed()
        stopRecord()
    }

    override fun onPause() {
        super.onPause()
        stopRecord()
    }

    fun stopRecord(){
        _isRecording = false
        _button?.text = "start"
        _record?.cancel(true)
    }

    fun doRecord(){
        _isRecording = true
        _button?.text = "stop"

        // AsyncTaskは使い捨て１回こっきりなので毎回作ります
        _record = Record()
        _record?.execute()
    }

    inner class Record : AsyncTask<Void, DoubleArray, Void>() {
        override fun doInBackground(vararg params: Void): Void? {
            // サンプリングレート。1秒あたりのサンプル数
            // （8000, 11025, 22050, 44100, エミュでは8kbじゃないとだめ？）
            val sampleRate = 8000

            // 最低限のバッファサイズ
            val minBufferSize = AudioRecord.getMinBufferSize(
                sampleRate,
                AudioFormat.CHANNEL_IN_MONO,
                AudioFormat.ENCODING_PCM_16BIT) * 2

            // バッファサイズが取得できない。サンプリングレート等の設定を端末がサポートしていない可能性がある。
            if(minBufferSize < 0){
                return null
            }

            val audioRecord = AudioRecord(
                MediaRecorder.AudioSource.MIC,
                sampleRate,
                AudioFormat.CHANNEL_IN_MONO,
                AudioFormat.ENCODING_PCM_16BIT,
                minBufferSize)

            val sec = 1
            val buffer: ShortArray = ShortArray(sampleRate * (16 / 8) * 1 * sec)

            audioRecord.startRecording()

            try {
                while (_isRecording) {
                    val readSize = audioRecord.read(buffer, 0, minBufferSize)

                    if (readSize < 0) {
                        break
                    }
                    if (readSize == 0) {
                        continue
                    }
                    if(buffer.isEmpty()){
                        break
                    }

                    if(buffer[0] < 1000 && buffer[0] > -1000){
                        t1.text=buffer[0].toString()
                        textView.text="ok"
                    }else{
                        t1.text=buffer[0].toString()
                        textView.text="no"
                    }
                    //_visualizer?.update(buffer, readSize)
                }
            } finally {
                audioRecord.stop()
                audioRecord.release()
            }

            return null
        }
    }
}